Consistent failures in the TestRespository unit test

### Environment
Device and OS: Ubuntu 22.04 Linux 

### Steps to reproduce
1. `make test-unit`

### Expected result

All tests passing

### Actual Result

Reports of consistent failures on specific machines.

### Visual Proof (screenshots, videos, text, etc)
```
utils.go:25: request: GET 127.0.0.1:36113/test.git/info/refs?service=git-receive-pack
utils.go:25: request: POST 127.0.0.1:36113/test.git/git-receive-pack
utils.go:25: request: GET 127.0.0.1:36113/test.git/info/refs?service=git-upload-pack
utils.go:25: request: POST 127.0.0.1:36113/test.git/git-upload-pack

[30;46m[30;46m NOTE [0m[0m [96m[96mFalling back to host 'git', failed to clone the repo "http://127.0.0.1:36113/test.git" with[0m[0m
[30;46m[30;46m      [0m[0m [96m[96mZarf: reference not found[0m[0m
utils.go:25: request: GET 127.0.0.1:36113/test.git/info/refs?service=git-upload-pack
utils.go:25: request: POST 127.0.0.1:36113/test.git/git-upload-pack
utils.go:25: request: GET 127.0.0.1:36113/test.git/info/refs?service=git-upload-pack
utils.go:25: request: GET 127.0.0.1:36113/test.git/info/refs?service=git-upload-pack
=== NAME  TestRepository
    repository_test.go:80: 
        	Error Trace:	/home/dev/work/zarf/src/internal/git/repository_test.go:80
        	Error:      	Received unexpected error:
        	            	packfile not found
        	Test:       	TestRepository
--- FAIL: TestRepository (0.05s)
FAIL
coverage: 28.0% of statements
FAIL	github.com/zarf-dev/zarf/src/internal/git	0.075s
```

### Severity/Priority

medium to high?

### Additional Context
Given the fallback to host git - this may be an issue with regards to host git or there is another error contributing to the fallback being executed.

--- a/src/internal/git/repository_test.go
+++ b/src/internal/git/repository_test.go
@@ -7,6 +7,7 @@ import (
 	"fmt"
 	"net/http"
 	"net/http/httptest"
+	"os"
 	"path/filepath"
 	"testing"
 
@@ -14,6 +15,7 @@ import (
 	"github.com/go-git/go-billy/v5/memfs"
 	"github.com/go-git/go-git/v5"
 	"github.com/go-git/go-git/v5/config"
+	"github.com/go-git/go-git/v5/plumbing"
 	"github.com/go-git/go-git/v5/plumbing/object"
 	"github.com/go-git/go-git/v5/storage/memory"
 	"github.com/stretchr/testify/require"
@@ -47,10 +49,15 @@ func TestRepository(t *testing.T) {
 
 	storer := memory.NewStorage()
 	fs := memfs.New()
-	initRepo, err := git.Init(storer, fs)
+	options := git.InitOptions{
+		DefaultBranch: plumbing.Main,
+	}
+	initRepo, err := git.InitWithOptions(storer, fs, options)
 	require.NoError(t, err)
+
 	w, err := initRepo.Worktree()
 	require.NoError(t, err)
+
 	filePath := "test.txt"
 	newFile, err := fs.Create(filePath)
 	require.NoError(t, err)
@@ -71,10 +78,17 @@ func TestRepository(t *testing.T) {
 		URLs: []string{repoAddress},
 	})
 	require.NoError(t, err)
+	t.Log(initRepo.Head())
 	err = initRepo.Push(&git.PushOptions{
 		RemoteName: "origin",
 	})
 	require.NoError(t, err)
+
+	// TODO: Is there a configuration that defines contents of HEAD that isn't read from ~/.gitconfig
+	// Force-write refs/heads/main ref to HEAD to disk - Matching the above reference and decoupling from host gitconfig
+	headFile := filepath.Join(cfg.Dir, "test.git", "HEAD")
+	err = os.WriteFile(headFile, []byte("ref: refs/heads/main\n"), 0644)
+	require.NoError(t, err, "Failed to write HEAD to disk")
 
 	repo, err := Clone(ctx, rootPath, repoAddress, false)
 	require.NoError(t, err)

--- a/src/internal/git/repository_test.go
+++ b/src/internal/git/repository_test.go
@@ -54,10 +54,8 @@ func TestRepository(t *testing.T) {
 	}
 	initRepo, err := git.InitWithOptions(storer, fs, options)
 	require.NoError(t, err)
-
 	w, err := initRepo.Worktree()
 	require.NoError(t, err)
-
 	filePath := "test.txt"
 	newFile, err := fs.Create(filePath)
 	require.NoError(t, err)
@@ -78,7 +76,6 @@ func TestRepository(t *testing.T) {
 		URLs: []string{repoAddress},
 	})
 	require.NoError(t, err)
-	t.Log(initRepo.Head())
 	err = initRepo.Push(&git.PushOptions{
 		RemoteName: "origin",
 	})

